src/porepy/applications/verification_setups/manu_flow_incomp_frac.py

```diff
@@ -477,188 +657,192 @@ def set_md_grid(self) -> None:
-        )
+        self.subdomain_darcy_flux_variable: str = "darcy_flux"
+        """Keyword to access the subdomain Darcy fluxes."""
 
-    def _plot_fracture_pressure(self):
-        sd_frac = self.mdg.subdomains()[1]
-        cc = sd_frac.cell_centers
-        p_num = self.sol.approx_frac_pressure
-        p_ex = self.sol.exact_frac_pressure
-        plt.plot(p_ex, cc[1], label="Exact", linewidth=3, alpha=0.5)
-        plt.plot(p_num, cc[1], label="MPFA", marker=".", markersize=5, linewidth=0)
-        plt.xlabel("Fracture pressure")
-        plt.ylabel("y-coordinate")
-        plt.legend()
-        plt.show()
+    def set_materials(self):
+        """Set material constants for the verification setup."""
+        super().set_materials()
 
-    # FIXME: Use sidegrids to separate left and right mortar fluxes
-    def _plot_interface_fluxes(self):
-        intf = self.mdg.interfaces()[0]
-        cc = intf.cell_centers
-        lmbda_num = self.sol.approx_intf_flux
-        lmbda_ex = self.sol.exact_intf_flux
-        plt.plot(lmbda_ex, cc[1], label="Exact", linewidth=3, alpha=0.5)
-        plt.plot(lmbda_num, cc[1], label="MPFA", marker=".", markersize=5, linewidth=0)
-        plt.xlabel("Interface flux")
-        plt.ylabel("y-coordinate")
-        plt.legend()
-        plt.show()
+        # Sanity checks to guarantee the validity of the manufactured solution
+        assert self.fluid.density() == 1
+        assert self.fluid.viscosity() == 1
+        assert self.fluid.compressibility() == 0
+        assert self.solid.permeability() == 1
+        assert self.solid.residual_aperture() == 1
+        assert self.solid.normal_permeability() == 0.5
 
-    def _plot_fracture_fluxes(self):
-        sd_frac = self.mdg.subdomains()[1]
-        fc = sd_frac.face_centers
-        q_num = self.sol.approx_frac_flux
-        q_ex = self.sol.exact_frac_flux
-        plt.plot(q_ex, fc[1], label="Exact", linewidth=3, alpha=0.5)
-        plt.plot(q_num, fc[1], label="MPFA", marker=".", markersize=5, linewidth=0)
-        plt.xlabel("Fracture Darcy flux")
-        plt.ylabel("y-coordinate")
-        plt.legend()
-        plt.show()
+        # Instantiate exact solution object after materials have been set
+        self.exact_sol = ManuIncompExactSolution()
 
-    def after_simulation(self) -> None:
-        """Method to be called after the simulation has finished"""
+    def after_nonlinear_convergence(
+        self, solution: np.ndarray, errors: float, iteration_counter: int
+    ) -> None:
+        """Method to be called after if solution has converged."""
 
-        # TODO: This is ugly. Is there a better way to do this?
-        # Compute Darcy fluxes and store them in pp.STATE
-        darcy_flux = self.darcy_flux(self.mdg.subdomains())
-        vals = darcy_flux.evaluate(self.equation_system).val
-        nf_rock = self.mdg.subdomains()[0].num_faces
+        # Store subdomain Darcy fluxes in pp.STATE
```

Comment:
Why?

Reply:
Simply to be able to access them in the data-saving stage. Do you have another suggestion?

Reply:
Could you compute them directly in data-saving stage?

Reply:
Sure.

Reply:
Done for all verifications.

Reply:
Does that mean that the secondary variable names should not be used? If yes, please purge them.

Reply:
Done.